docs/sessions.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: default
+---
+
+State and Sessions in Suave
+===========================
+
+While stateless web applications are all the rage, there are still times where state needs to be preserved, either across requests in a session, or simply among `WebPart`s as the request/response pipeline is composed.  In these cases, Suave has you covered!  We'll look at three aspects of session cookies first, then how we can construct state among `WebPart`s.
+
+Cookie State Session Storage
+----------------------------
+
+Suave provides storage interfaces for cookies in the `Suave.State.CookieStateStore` module.  The `statefulForSession` WebPart can be composed to make a path session-aware.  From there, the `HttpContext.state` function extracts the state information, and the `get` and `set` functions on the resulting `StateStore` object can be used to manipulate the contents of the state tracked by the session.
+
+{% highlight fsharp %}
+open Suave
+open Suave.State.CookieStateStore
+
+/// string -> 'T -> WebPart
+let setSessionValue key value = context (fun ctx ->
+  match ctx |> HttpContext.state with
+  | Some state -> state.set key value
+  | _ -> never // fail)
+
+/// HttpContext -> string -> 'T option
+let getSessionValue ctx key =
+  match ctx |> HttpContext.state with
+  | Some state -> state.get key
+  | _ -> None
+
+/// HttpContext -> string -> string
+let getStringSessionValue ctx key = 
+  match getSessionValue ctx key with
+  | Some value -> string value
+  | _ -> ""
+
+/// WebPart
+let cookieYes = warbler (fun ctx -> OK (getStringSessionValue ctx "test"))
```

Comment:
You could use `context` here for clarity and code completion support in the lamda body.